profiling/src/lib.rs

```diff
@@ -908,7 +908,14 @@ fn notify_trace_finished(local_root_span_id: u64, span_type: Cow<str>, resource:
 /// interrupt_count.
 fn interrupt_function(execute_data: *mut zend::zend_execute_data) {
     REQUEST_LOCALS.with(|cell| {
-        let mut locals = cell.borrow_mut();
+        // try to borrow and bail out if not successful
+        let mut locals = match cell.try_borrow_mut() {
+            Ok(locals) => locals,
+            Err(_) => {
+                return;
+            }
+        };
+
```

Comment:
@morrisonlevi can you check this change?
The reason to this change was that the `REQUEST_LOCALS` could have already been borrowed in `datadog_php_profiling::timeline::ddog_php_prof_gc_collect_cycles` which wrapped the call to the original `gc_collect_cylces()` function. I changed that part to not have the `REQUEST_LOCALS` borrowed anymore when making the call to `gc_collect_cycles()`, so we might not need this anymore.